### ISSUE TITLE
Averages don't take into account milliseconds

### ISSUE THREAD

**Issue by ericfrederich** [sdispater/pendulum#137]
Averages seem to just copy the millisecond portion instead of averaging them.

For two instances of Pendulum `p1` and `p2` I believe it should hold that `p1.average(p2) == p2.average(p1)`

This is not currently the case.  In fact, averaging two times that are only 1 second apart will yield one of the source times rather than a midpoint.

```Python
import pendulum
import time

p1 = pendulum.now()
time.sleep(1)
p2 = pendulum.now()

a1 = p1.average(p2)
a2 = p2.average(p1)

print(p1)
print(a1)
print(a2)
print(p2)
print(a1 == a2)
print(p1 == a1)
print(p2 == a2)
```
**example output:**
```
2017-08-23T12:01:32.809279-04:00
2017-08-23T12:01:32.809279-04:00
2017-08-23T12:01:33.821634-04:00
2017-08-23T12:01:33.821634-04:00
False
True
True
```

**Comment by sdispater** [comment]
Yes, you are definitely right.

I just checked the code and `average()` only averages at the `seconds` level and not microseconds.

It's an easy fix so it should not be too long before it makes it to the next minor release

**Comment by ericfrederich** [comment]
@sdispater, great!  Add a test too if you can.  I currently have datetime code that recursively bisects intervals.  I'd like to switch to pendulum but obviously cannot until this is fixed.

**Comment by ericfrederich** [comment]
Pinging @sdispater ... I wrote a fix and some tests.

Things in 2.0 branch are refactored so it's not a clean merge.  I can merge conflicts and get 2.0 branch up to date once this is merged into master.

**Comment by ericfrederich** [comment]
Look good @sdispater ?

**Comment by sdispater** [comment]
@ericfrederich Yes, your PR fixed this. However I will close this once I publish the next release.
